commit de56aa686a214ff541273d6a2d8d5b3db205de2e
Author: Niklas Cassel <niklas.cassel@axis.com>
Date:   Sat Feb 25 01:17:53 2017 +0100

    locking/rwsem: Fix down_write_killable() for CONFIG_RWSEM_GENERIC_SPINLOCK=y
    
    commit 17fcbd590d0c3e35bd9646e2215f86586378bc42 upstream.
    
    We hang if SIGKILL has been sent, but the task is stuck in down_read()
    (after do_exit()), even though no task is doing down_write() on the
    rwsem in question:
    
      INFO: task libupnp:21868 blocked for more than 120 seconds.
      libupnp         D    0 21868      1 0x08100008
      ...
      Call Trace:
      __schedule()
      schedule()
      __down_read()
      do_exit()
      do_group_exit()
      __wake_up_parent()
    
    This bug has already been fixed for CONFIG_RWSEM_XCHGADD_ALGORITHM=y in
    the following commit:
    
     04cafed7fc19 ("locking/rwsem: Fix down_write_killable()")
    
    ... however, this bug also exists for CONFIG_RWSEM_GENERIC_SPINLOCK=y.
    
    Signed-off-by: Niklas Cassel <niklas.cassel@axis.com>
    Signed-off-by: Peter Zijlstra (Intel) <peterz@infradead.org>
    Cc: <mhocko@suse.com>
    Cc: Andrew Morton <akpm@linux-foundation.org>
    Cc: Linus Torvalds <torvalds@linux-foundation.org>
    Cc: Niklas Cassel <niklass@axis.com>
    Cc: Paul E. McKenney <paulmck@linux.vnet.ibm.com>
    Cc: Peter Zijlstra <peterz@infradead.org>
    Cc: Thomas Gleixner <tglx@linutronix.de>
    Fixes: d47996082f52 ("locking/rwsem: Introduce basis for down_write_killable()")
    Link: http://lkml.kernel.org/r/1487981873-12649-1-git-send-email-niklass@axis.com
    Signed-off-by: Ingo Molnar <mingo@kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1459_de56aa686a214ff541273d6a2d8d5b3db205de2e.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1459_de56aa686a214ff541273d6a2d8d5b3db205de2e.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
